pkg/abi/linux/fuse.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2018 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package linux
+
+// +marshal
+type FUSEOpcode uint32
+
+// +marshal
+type FUSEOpID uint64
+
+// Opcodes for FUSE operations. Analogous to the opcodes in include/linux/fuse.h.
+const (
+	FUSE_LOOKUP FUSEOpcode = iota + 1
+	FUSE_FORGET /* no reply */
+	FUSE_GETATTR
+	FUSE_SETATTR
+	FUSE_READLINK
+	FUSE_SYMLINK
+	_
```

Comment:
We might need hosts to confirm this. I found many explicit const

https://github.com/google/gvisor/blob/cfd30665c1d857f20dd05e67c6da6833770e2141/pkg/abi/linux/netfilter.go

https://github.com/google/gvisor/search?l=Go&q=const

Reply:
No strong preferences either way, it is acceptable to use iota here since the values are (mostly) sequential.

However when the actual value of an enum is important (i.e. they have a semantic meaning, and aren't just unique numbers to each symbol), I personally prefer defining them explicitly.

Reply:
Agree. Reason for my asking is, the unix has defined non-sequential opt_code like this: https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/fuse.h#L470 Being explicit here could make the future expansion slightly easier but no strong preferences.